cmd/savepcdfiles/main.go

```diff
@@ -93,14 +93,19 @@ func savePCDFiles(ctx context.Context, port int, lidarComponent config.Component
 
 	// Run loop
 	for {
-		if !utils.SelectContextOrWait(ctx, time.Second) {
+		if !utils.SelectContextOrWait(ctx, 10*time.Millisecond) {
 			return multierr.Combine(ctx.Err(), myRobot.Close(ctx))
 		}
 
 		pc, err := rplidar.NextPointCloud(ctx)
 		if err != nil {
 			return multierr.Combine(err, myRobot.Close(ctx))
 		}
-		logger.Infow("scanned", "pointcloud_size", pc.Size())
+
+		if pc != nil {
+			logger.Infow("scanned", "pointcloud_size", pc.Size())
+		} else {
+			logger.Infow("warning error reading pcd, skipping")
+		}
```

Comment:
Related to the discussion below, does the else clause get executed at all, given that we're already checking for an error above that?